dpctl/apis/include/dpctl4pybind11.hpp

```diff
@@ -842,20 +842,29 @@ class usm_ndarray : public py::object
 
     int get_ndim() const
     {
-        PyUSMArrayObject *raw_ar = this->usm_array_ptr();
+        PyUSMArrayObject *raw_ar = usm_array_ptr();
 
         auto const &api = ::dpctl::detail::dpctl_capi::get();
         return api.UsmNDArray_GetNDim_(raw_ar);
     }
 
     const py::ssize_t *get_shape_raw() const
     {
-        PyUSMArrayObject *raw_ar = this->usm_array_ptr();
+        PyUSMArrayObject *raw_ar = usm_array_ptr();
 
         auto const &api = ::dpctl::detail::dpctl_capi::get();
         return api.UsmNDArray_GetShape_(raw_ar);
     }
 
+    const std::vector<py::ssize_t> get_shape_vector() const
```

Comment:
```suggestion
    std::vector<py::ssize_t> get_shape_vector() const
```

The returned vector is a copy already, there should be no harm in modifying it.